env0/resource_configuration_variable.go

```diff
@@ -61,7 +63,7 @@ func resourceConfigurationVariable() *schema.Resource {
 				Type:          schema.TypeString,
 				Description:   "create the variable under this environment, not globally",
 				Optional:      true,
-				ConflictsWith: []string{"template_id", "project_id"},
+				ConflictsWith: []string{"template_id", "project_id", "is_required", "is_read_only"},
```

Comment:
Why these flags conflictWith environment_id?

Reply:
it's because they are only relevant for scopes that have lower scopes.
`environment` is the lowest scope, so it doesn't make sense to make its conf variable `is_required` or `read_only` for lower scopes because there aren't such.